src/main/scala/org/bblfsh/client/v2/BblfshClient.scala

```diff
@@ -0,0 +1,161 @@
+package org.bblfsh.client.v2
+
+import java.nio.ByteBuffer
+
+import com.google.protobuf.ByteString
+import com.google.protobuf.duration.Duration
+import io.grpc.ManagedChannelBuilder
+import gopkg.in.bblfsh.sdk.v2.protocol.driver.{ParseRequest, ParseResponse, DriverGrpc, SupportedLanguagesRequest, SupportedLanguagesResponse, VersionRequest, VersionResponse, DriverHostGrpc}
+
+import org.bblfsh.client.v2.libuast.Libuast
+
+
+class BblfshClient(host: String, port: Int, maxMsgSize: Int) {
+  private val DEFAULT_TIMEOUT_SEC = 5
+
+  private val channel = ManagedChannelBuilder
+    .forAddress(host, port)
+    .usePlaintext(true)
+    .maxInboundMessageSize(maxMsgSize)
+    .build()
+
+  private val stub = DriverGrpc.blockingStub(channel)
+  private val stubInfo = DriverHostGrpc.blockingStub(channel)
+
+  /**
+    * Parses file with a given name and content using 
+    * the provided timeout.
+    *
+    * @param name file name
+    * @param content file content
+    * @param lang (optional) language to parse, default auto-detect \w enry
+    * @param timeout (disabled) bblfsh request timeout, seconds
+    *                Right now this does not have any effect in v2.
+    * @return UAST in parse response.
+    */
+  def parseWithOptions(
+    name: String,
+    content: String,
+    lang: String,
+    timeout: Long
+  ): ParseResponse = {
+    // FIXME(bzz): make timout work in v2 again
```

Comment:
Oh, I see, so it's gone from the request now and should be set by other means. I guess it's good for the client since now it will also know that we have a timeout for the RPC call.

Reply:
Yup, and to clarify the nature of this change as well as migration part for the clients as well as adjust up the API - there is a separate todo item in the umbrella issue.